### PR TITLE
Fixes #10413 - Drop unattended: false mode

### PR DESCRIPTION
**Pull request by tbrisker**
Creating as draft, this should only be merged after 3.2 is branched. 
@ezr-ondrej has agreed to rebase it when the time comes.
Feel free to squash on merge, split into multiple commits to make review easier.

### CODE REVIEW

File: app/models/host_info_providers/static_info.rb [hunk 1]
```diff
@@ -8,12 +8,12 @@ def host_info
       param["fqdn"] = host.fqdn unless host.fqdn.nil?
       param["hostgroup"] = host.hostgroup.to_label unless host.hostgroup.nil?
       param["comment"] = host.comment if host.comment.present?
+      param["root_pw"] = host.root_pass unless (!host.operatingsystem.nil? && host.operatingsystem.password_hash == 'Base64')
```
Comment (ekohl) [inline app/models/host_info_providers/static_info.rb:11]: I know you're just moving it, but I found this horribly complex to read. Trying to simplify it (removing some negations)
```suggestion
      param["root_pw"] = host.root_pass if (host.operatingsystem.nil? || host.operatingsystem.password_hash != 'Base64')
```

And is that then the same as:

```suggestion
      param["root_pw"] = host.root_pass if host.operatingsystem&.password_hash != 'Base64'
```

File: app/models/nic/base.rb [hunk 2]
```diff
@@ -225,11 +225,8 @@ def children_mac_addresses
       []
     end
 
-    # we don't consider host as managed if we are in non-unattended mode
-    # in which case host managed? flag can be true but we should consider
-    # everything as unmanaged
     def host_managed?
-      host&.managed? && SETTINGS[:unattended]
+      host&.managed?
```
Comment (ekohl) [inline app/models/nic/base.rb:229]: Given this simplicity, should we deprecate it and suggest to inline this?

File: config/settings.rb [hunk 3]
```diff
@@ -12,12 +12,12 @@
 SETTINGS.deep_merge!(Foreman::EnvSettingsLoader.new.to_h)
 
 # Force setting to true until all code using it is removed
-[:locations_enabled, :organizations_enabled].each do |setting|
+[:locations_enabled, :organizations_enabled, :unattended].each do |setting|
   SETTINGS[setting] = true
```
Comment (ekohl) [inline config/settings.rb:16]: I wonder if we could implement a special Hash type in development mode that generates a deprecation warning when these settings are used.

**Review by ekohl — DISMISSED** [review]

File: app/models/host_status/build_status.rb [hunk 4]
```diff
@@ -56,10 +56,6 @@ def to_status(options = {})
       end
     end
 
-    def relevant?(options = {})
```
Comment (ezr-ondrej) [inline app/models/host_status/build_status.rb:59]: Should we just deprecate and return `true` in the meantime?

**Review by ezr-ondrej — COMMENTED** [review]
Nice :+1:
Just one nit

File: app/models/host_status/build_status.rb (hunk 4 above)
Comment (tbrisker) [reply app/models/host_status/build_status.rb:59]: We do return true, from the base class definition. Other status have different definitions for `relevant?`

File: app/models/host_info_providers/static_info.rb (hunk 1 above)
Comment (tbrisker) [reply app/models/host_info_providers/static_info.rb:11]: I agree the current condition is confusing... However, the current implementation actually includes the `root_pw` if the os is `nil`. Not sure if that is correct behavior or not, but I'd rather not modify it in this PR.

File: config/settings.rb (hunk 3 above)
Comment (tbrisker) [reply config/settings.rb:16]: Probably not worth it for the once in a blue moon when we drop one of these, and by forcing them to true we avoid breaking any code that might still check them

File: app/models/nic/base.rb (hunk 2 above)
Comment (tbrisker) [reply app/models/nic/base.rb:229]: perhaps, but please not in this PR.

**Review by ezr-ondrej — APPROVED** [review]
Great clean up :+1:
This should get an upgrade warning and we should discuss it on the maintainers call before merging, but IMHO it is ready to go in :+1:

**Comment by ezr-ondrej** [comment]
Upgrade warning: https://github.com/theforeman/theforeman.org/pull/1964

**Comment by ezr-ondrej** [comment]
I'll go ahead and merge this, we have the upgrade warning PR open and I'm gonna mark this as a headline feature.

**Comment by tbrisker** [comment]
🥳 Thanks for taking this to the finish line @ezr-ondrej ! I've wanted to make this change for years :)

**Comment by ezr-ondrej** [comment]
The pleasure was mine, I love simplifying code :heart: 

> I've wanted to make this change for years

And it would not happen without you, so big thanks ! :+1:

**Pull request merged**